### DISCUSSION TITLE
getContentSize() bug

### DISCUSSION THREAD

**Discussion by rudiHammad** [axmolengine/axmol#1237] [Q&A] [answered]
Hi,
I am new to axmol but I had a project in cocos2dx that I am migrating and I am noticing some issues I don't think I had before.
For instace, I create a square spritge of an image with a resolution of 50 * 50 and they a check the content size as usual with

auto test = ax::Sprite::create("redRect.png");
addChild(test);
auto size = test->getContentSize();
ax::log("%f, %f <<<<<<<<", size.width, size.height);

And the results are 62.5 width and 62.5 height. I can't figure out why is that. Is it a but or am I missing something?
Just in case you might think the image is not actually 50*50 here are the details

![Capture](https://github.com/axmolengine/axmol/assets/25636976/67b0f0b8-4784-4833-b8b9-1509c4c5e9f5)

**Comment by halx99** [comment, marked as answer]
The contentSize of sprite affected by contentScaleFacotor, not the image pixel wide, if you want get the image pixel wide and height, use follow code:

```cpp
auto test = ax::Sprite::create("redRect.png");
auto texture = test->getTexture();
int imageWidth = texture->getPixelsWide();
int imageHeight = texture->getPixelsHigh();
```

**Reply by rudiHammad** [reply]
okey thanks. But what I don't understand is that I didn't change thing in the scale. It is just an 50*50 image, with nothing scaled.

**Reply by rh101** [reply]
Look at your project's `AppDelegate.cpp` file, specifically the calls to `setContentScaleFactor`. Put a breakpoint on the calls to that function, and you will have your answer as to why the scale factor is equal to `1.25`.

If you don't want to use scaling, then just get rid of those setContentScaleFactor calls, or just make sure it's called with a value of 1.  In this case you would need to make sure you set the correct `ResolutionPolicy` for your needs when calling `glView->setDesignResolutionSize`.

**Reply by rudiHammad** [reply]
oh my god. Thanks. 
I spent 3 hours reviewing custom tools i wrote because all my animations where out of frame. That fixed it.
Thanks for the quick response.
